### DISCUSSION TITLE
Is omniauth-rails_csrf_protection still required for OmniAuth 2.0?

### DISCUSSION THREAD

**Discussion by nathan-gusto** [omniauth/omniauth#1046] [Q&A] [answered]
Hi there!

I read through the [upgrade guide to 2.0](https://github.com/omniauth/omniauth/wiki/Upgrading-to-2.0#rails) and would appreciate some clarification on whether `omniauth-rails_csrf_protection` is still required after upgrading `omniauth`. 

The update section noted that if we're already using `omniauth-rails_csrf_protection` then we'd be able to upgrade to 2.0. What happens if we're _not_ using it? Do we need to add the gem or the custom implementation for `omniauth 2.0` to work properly?

Thanks!

**Comment by BobbyMcWho** [comment, marked as answer]
Hi Nathan!

The gem you linked is indeed the easiest way to set the authenticity token callback to match Rails. Whether you use that gem or implement your own, you will have to do _something_ as omniauth tries to not include rails specific code in the gem.

**Reply by nathan-gusto** [reply]
Thanks so much @BobbyMcWho for the reply and all of your work for maintaining this gem!
